document-generation/generate-document.ts

```diff
@@ -6,64 +6,68 @@ import inputOutputLogger from "@middy/input-output-logger";
+import { INTERNAL_SERVER_ERROR } from "../utils/errors";
 import { LoggingContextMiddleware } from "../utils/middleware/logging-context-middleware";
 import { errorHandlingMiddleware } from "../utils/middleware/error-handling-middleware";
 import JSONErrorHandlerMiddleware from "middy-middleware-json-error-handler";
 import validator from "@middy/validator";
 import xssSanitizer from "../utils/middleware/xss-sanitizer";
 import { wrapSchema } from "../utils/middleware/schema-wrapper";
 import { generateDocument } from "./chromium";
+import { generateIGCEDocument } from "./igce-document";
 import {
   generateDocumentSchema,
   RequestEvent,
   GenerateDocumentRequest,
   DocumentType,
+  TemplatePaths,
+  IndependentGovernmentCostEstimate,
 } from "../models/document-generation";
 import * as fs from "fs";
 import handlebars from "handlebars";
 import juice from "juice";
-import {
-  formatDuration,
-  formatGroupAndClassification,
-  counter,
-  countSections,
-  formatAwardType,
-} from "./handlebarUtils/helpers";
+import { formatDuration, formatGroupAndClassification, counter, countSections, formatAwardType } from "./utils/helpers";
 
 async function baseHandler(event: RequestEvent<GenerateDocumentRequest>): Promise<ApiBase64SuccessResponse> {
   const { documentType } = event.body;
+  logger.info("Generating document", { documentType });
+  const documentTemplatePaths: TemplatePaths = {
+    [DocumentType.DESCRIPTION_OF_WORK]: {
+      html: "/opt/dow-template.html",
+      css: "/opt/dow-style.css",
+    },
+    [DocumentType.INDEPENDENT_GOVERNMENT_COST_ESTIMATE]: {
+      excel: "/opt/igce-template.xlsx",
+    },
+  };
+
   switch (documentType) {
     case DocumentType.DESCRIPTION_OF_WORK:
-      return generatePdf(event);
+      return generatePdf(event, documentTemplatePaths);
     case DocumentType.INDEPENDENT_GOVERNMENT_COST_ESTIMATE:
-      return generateXlsx();
+      return generateXlsx(event, documentTemplatePaths);
     default:
       return new ValidationErrorResponse(`Invalid document type: "${documentType}"`, {
         cause: `Invalid document type "${documentType}" provided. Please provide a valid document  type.`,
       });
   }
 }
 
-async function generatePdf(event: RequestEvent<GenerateDocumentRequest>): Promise<ApiBase64SuccessResponse> {
+async function generatePdf(
+  event: RequestEvent<GenerateDocumentRequest>,
+  documentTemplatePaths: TemplatePaths
+): Promise<ApiBase64SuccessResponse> {
   const { documentType, templatePayload } = event.body;
+  let html = "";
+  let css = "";
 
-  const documentTemplatePaths = {
-    [DocumentType.DESCRIPTION_OF_WORK as string]: {
-      html: "/opt/dow-template.html",
-      css: "/opt/dow-style.css",
-    },
-  };
+  if (documentType === DocumentType.DESCRIPTION_OF_WORK) {
+    // get files to generate documents
+    html = fs.readFileSync(documentTemplatePaths[documentType].html, "utf-8");
+    css = fs.readFileSync(documentTemplatePaths[documentType].css, "utf-8");
+  }
```

Comment:
This is really unfortunate for a situation that (currently) can never be false. I'd almost rather break this out into it's own function; and perhaps with a `default` that throws an error:

```ts
let html = "";
let css = "";
switch (documentType) {
  case DocumentType.DESCRIPTION_OF_WORK:
    html = fs.readFileSync(documentTemplatePaths[documentType].html, "utf-8");
    css = fs.readFileSync(documentTemplatePaths[documentType].css, "utf-8");
  default:
    throw new Error(`Unsupported PDF generation type: "${documentType}"`
}
```

Reply:
btw, I know the linter will hate this (there's only 1 `case`) but I'd be cool with an `eslint-disable` for this.

Reply:
yeah, that's true. I think I was trying to keep Typescript happy otherwise it complains about the `html` or `css` properties.

I'll break it into it's own function.